test/jdk/java/lang/reflect/Proxy/ClassRestrictions.java

```diff
@@ -52,129 +58,90 @@ interface Bashful {
         void foo();
     }
 
-    public static final String nonPublicIntrfaceName = "java.util.zip.ZipConstants";
-
-    public static void main(String[] args) {
+    static Stream<Arguments> proxyInterfaces() {
+        return Stream.of(
+                /*
+                 * All of the Class objects in the interfaces array must represent
+                 * interfaces, not classes or primitive types.
+                 */
+                Arguments.of(new Class<?>[] { Object.class }, "proxy class created with java.lang.Object as interface"),
+                Arguments.of(new Class<?>[] { Integer.TYPE }, "proxy class created with int.class as interface"),
+                Arguments.of(new Class<?>[] { Bar.class, Bar.class }, "proxy class created with repeated interfaces"),
+                /*
+                 * No two interfaces may each have a method with the same name and
+                 * parameter signature but different return type.
+                 */
+                Arguments.of(new Class<?>[] { Bar.class, Baz.class }, "proxy class created with conflicting methods")
+        );
+    }
 
-        System.err.println(
-            "\nTest of restrictions on parameters to Proxy.getProxyClass\n");
+    /*
+     * Test valid interfaces
+     */
+    @ParameterizedTest
+    @MethodSource("proxyInterfaces")
+    void testForName(Class<?>[] interfaces, String message) {
+        ClassLoader loader = ClassRestrictions.class.getClassLoader();
+        try {
+            var proxyClass = Proxy.getProxyClass(loader, interfaces);
+            throw new Error(message);
+        } catch (IllegalArgumentException e) {
+            System.err.println(e.getMessage());
+            // assume exception is for intended failure
```

Comment:
The "throw new Error(message)" should be replaced with fail(message) or use assertThrows(IllegalArgument.class, () -> Proxy.getProxyClass(...)).

Reply:
Will do.  I should clean up more of the existing code.